blockchain/ethereum.go

```diff
@@ -85,6 +85,22 @@ func newEVMClient(networkSettings EVMNetwork) (EVMClient, error) {
 	ec.gasStats = NewGasStats(ec.ID)
 	go ec.newHeadersLoop()
 
+	// Check if the chain supports EIP-1559
+	// https://eips.ethereum.org/EIPS/eip-1559
+	if _, err = cl.SuggestGasTipCap(context.Background()); err != nil {
+		if strings.Contains(err.Error(), "unknown field") ||
+			strings.Contains(err.Error(), "does not exist") ||
+			strings.Contains(err.Error(), "is not available") {
+			ec.supportsEIP1559 = false
+			log.Debug().Msg("Chain does not support EIP-1559 transactions, using legacy")
```

Comment:
Can we not simply take it as a network input?

Reply:
We can go that way too, though I think we'd need to do

1. Add another config field
2. Add another method to the EVMClient interface
3. Deal with possible cases of interface shadowing, which can be frustrating

Reply:
The network config seems cleaner and robust rather than calling the function and trying to parse the error message.